packages/pds/src/actor-store/index.ts

```diff
@@ -83,44 +53,46 @@ export class ActorStore {
   }
 
   async keypair(did: string): Promise<Keypair> {
-    const got = await this.keyCache.fetch(did)
-    if (!got) {
-      throw new InvalidRequestError('Keypair not found', 'NotFound')
-    }
-    return got
+    const { keyLocation } = await this.getLocation(did)
+    const privKey = await fs.readFile(keyLocation)
+    return crypto.Secp256k1Keypair.import(privKey)
   }
 
-  async db(did: string): Promise<ActorDb> {
-    const got = await this.dbCache.fetch(did)
-    if (!got) {
+  async openDb(did: string): Promise<ActorDb> {
+    const { dbLocation } = await this.getLocation(did)
+    const exists = await fileExists(dbLocation)
+    if (!exists) {
       throw new InvalidRequestError('Repo not found', 'NotFound')
     }
-    return got
-  }
 
-  async reader(did: string) {
-    const [db, keypair] = await Promise.all([this.db(did), this.keypair(did)])
-    return createActorReader(did, db, keypair, this.resources)
+    return getDb(dbLocation, this.cfg.disableWalAutoCheckpoint)
   }
 
   async read<T>(did: string, fn: ActorStoreReadFn<T>) {
-    const reader = await this.reader(did)
-    return fn(reader)
+    const keypair = await this.keypair(did)
```

Comment:
I believe this keypair is used with `ActorStoreReader.transact`, which may not be used anywhere.  So might be able to nix it?

Reply:
yup good call 👍